rust/src/detect/iprep.rs

```diff
@@ -84,7 +84,7 @@ pub fn detect_parse_iprep(i: &str) -> IResult<&str, DetectIPRepData> {
     let (i, name) = take_while(is_alphanumeric_or_slash)(i)?;
     // copy as to have final zero
     let namez = CString::new(name).unwrap();
-    let cat = unsafe { SRepCatGetByShortname(namez.as_ptr() as *const i8) };
+    let cat = unsafe { SRepCatGetByShortname(namez.as_ptr()) };
```

Comment:
This gives errors on my arm builders:
```
   Compiling suricata v7.0.1-dev (/builds/inliniac/suricata-ci/suricata/rust)
error[E0308]: mismatched types
  --> src/detect/iprep.rs:87:46
   |
87 |     let cat = unsafe { SRepCatGetByShortname(namez.as_ptr()) };
   |                        --------------------- ^^^^^^^^^^^^^^ expected `*const i8`, found `*const u8`
   |                        |
   |                        arguments to this function are incorrect
   |
   = note: expected raw pointer `*const i8`
              found raw pointer `*const u8`
note: function defined here
  --> src/detect/iprep.rs:74:12
   |
74 |     pub fn SRepCatGetByShortname(name: *const i8) -> u8;
   |            ^^^^^^^^^^^^^^^^^^^^^
For more information about this error, try `rustc --explain E0308`.
error: could not compile `suricata` (lib) due to previous error


   Compiling suricata v7.0.1-dev (/builds/inliniac/suricata-ci/suricata/rust)
error[E0308]: mismatched types
  --> src/detect/iprep.rs:87:46
   |
87 |     let cat = unsafe { SRepCatGetByShortname(namez.as_ptr()) };
   |                        --------------------- ^^^^^^^^^^^^^^ expected `i8`, found `u8`
   |                        |
   |                        arguments to this function are incorrect
   |
   = note: expected raw pointer `*const i8`
              found raw pointer `*const u8`
note: function defined here
  --> src/detect/iprep.rs:74:12
   |
74 |     pub fn SRepCatGetByShortname(name: *const i8) -> u8;
   |            ^^^^^^^^^^^^^^^^^^^^^
For more information about this error, try `rustc --explain E0308`.
error: could not compile `suricata` due to previous error
```
Not sure why the 2 errors are different, perhaps rustc version differences.

Reply:
IIRC C type `char` is unsigned in arm.

Reply:
Indeed